llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -379,33 +381,38 @@ Instruction *emitSpecConstantComposite(Type *Ty,
 /// encountered scalars and assigns them IDs (or re-uses existing ones).
 Instruction *emitSpecConstantRecursiveImpl(Type *Ty, Instruction *InsertBefore,
                                            SmallVectorImpl<unsigned> &IDs,
-                                           unsigned &Index) {
+                                           unsigned &Index,
+                                           Constant *DefaultValue) {
   if (!Ty->isArrayTy() && !Ty->isStructTy() && !Ty->isVectorTy()) { // Scalar
     if (Index >= IDs.size()) {
       // If it is a new specialization constant, we need to generate IDs for
       // scalar elements, starting with the second one.
       IDs.push_back(IDs.back() + 1);
     }
-    return emitSpecConstant(IDs[Index++], Ty, InsertBefore);
+    return emitSpecConstant(IDs[Index++], Ty, InsertBefore, DefaultValue);
   }
 
   SmallVector<Instruction *, 8> Elements;
-  auto LoopIteration = [&](Type *Ty) {
+  auto LoopIteration = [&](Type *Ty, unsigned LocalIndex) {
```

Comment:
Not clear to me, why `LocalIndex` is needed?

Reply:
It is used to understand which item of a composite/aggregate type should be used as a default value on each iteration.